File: JavaApp/src/main/java/com/charlezz/javaapp/feature/local/UserModule.java
package com.charlezz.javaapp.feature.local;

import android.view.LayoutInflater;

import androidx.databinding.DataBindingUtil;

import com.charlezz.javaapp.R;
import com.charlezz.javaapp.databinding.FragmentUserBinding;
import com.charlezz.javaapp.di.FragmentScope;
import com.charlezz.javaapp.util.CommonDataBindingComponent;

import dagger.Module;
import dagger.Provides;

@Module
public  class UserModule {
    @Provides
    @FragmentScope
    FragmentUserBinding provideBinding(UserFragment fragment, CommonDataBindingComponent commonDataBindingComponent){
        return DataBindingUtil.inflate(LayoutInflater.from(fragment.getContext()), R.layout.fragment_user, null, false, commonDataBindingComponent);
    }

}

File: KotlinApp/src/main/java/com/charlezz/finalarchitecture/feature/local/UserModule.kt
package com.charlezz.finalarchitecture.feature.local

import androidx.databinding.DataBindingUtil
import com.charlezz.finalarchitecture.R
import com.charlezz.finalarchitecture.databinding.ActivityUserBinding
import com.charlezz.finalarchitecture.di.ActivityScope
import dagger.Module
import dagger.Provides

@Module
class UserModule {

    @Provides
    @ActivityScope
    fun provideActivityPersonBinding(activity: UserActivity): ActivityUserBinding =
            DataBindingUtil.setContentView(activity, R.layout.activity_user)


}